api/writeAPIBlocking.go

```diff
@@ -80,18 +81,25 @@ func (w *writeAPIBlocking) write(ctx context.Context, line string) error {
 }
 
 func (w *writeAPIBlocking) WriteRecord(ctx context.Context, line ...string) error {
-	if len(line) > 0 {
-		var sb strings.Builder
-		for _, line := range line {
-			b := []byte(line)
-			b = append(b, 0xa)
-			if _, err := sb.Write(b); err != nil {
-				return err
-			}
+	if len(line) == 0 {
+		return nil
+	}
+	if len(line) == 1 {
```

Comment:
empty lines are ignored by InfluxDB, adding extra new line is not required, it could simply `return w.write(ctx, line[0])` 

https://github.com/influxdata/influxdb/blob/11c00813f106062829759a97673c7c945d5859ca/models/points.go#L369

Reply:
The existing implementation assumes there are no terminating newlines on the passed records; it's necessary to add one when missing to keep the existing behavior.  Also, not adding a newline causes the tests to fail, and keeps this consistent with what's produced by line-protocol, see influxdata/line-protocol#52.

Reply:
BTW: That existing behavior is verified by the tests, which is why it isn't the way you suggested....